lib/AST/GenericSignature.cpp

```diff
@@ -106,11 +106,11 @@ std::string GenericSignature::gatherGenericParamBindingsText(
     result += gp->getName().str();
     result += " = ";
 
-    auto found = substitutions.find(canonGP);
-    if (found == substitutions.end())
+    auto type = substitutions(canonGP);
```

Comment:
It is slightly more correct to do gp.subst(substitutions). Then you don't have to canonicalize it either. But feel free to merge this first and do this later